### ISSUE TITLE
Creating the Scope Conditions FIle

### ISSUE THREAD

**Issue by jordan-roberts** [jordan-roberts/International-Crisis-Dynamics#1]
To finish the scope conditions file:

1) Someone needs to improve the generation of the "crisisName" variable.  It is made by combing strings together with paste: "CRIS" + "IS N" + "AME" "EXAM" + "PLE".  While the separator is currently set to empty, this does not always produce the desired effect.  The above strings become "CRISIS NAMEEXAMPE" because there is no space character to separate NAME and EXAMPLE since the end of the word NAME is also the end of a string segment, or "cluster."

2) Someone needs to merge the SysScope and ActScope dataframes created at the end.  It should be a 1-to-many merge of some kind.

3) Someone needs to read the case summaries for cases in which the SysScope$mediate == 4, and determine the identity of the primary mediator in the crisis.  A column with the country ID for this mediator (and another code if the mediator is not a state actor) needs to be cbinded to the SysScope dataframe sometime before it is merged with ActScope.

If you decided you take one or more these on, say so here so we're not duplicating efforts.

**Comment by haoliuhoward** [comment]
I can do (2). But I can switch to (3) if Jordan thinks my experience with ICB should help on (3). And because there are many cases and mediators may not be explicitly stated in the summaries, it would be better to have multiple people to read the summaries to increase the speed.

**Comment by hail2thief** [comment]
I can do (1). Are we just working off the dataset available online or is
there a more recent one that is yet to be released?

On Thu, Sep 24, 2015 at 12:17 AM, Howard Liu <notifications@github.com>
wrote:

> I can do (2). But I can switch to (3) if Jordan thinks my experience with
> ICB should help on (3). And because there are many cases and mediators may
> not be explicitly stated in the summaries, it would be better to have
> multiple people to read the summaries to increase the speed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jordan-roberts/International-Crisis-Dynamics/issues/1#issuecomment-142807172>
> .
>



-- 
Best,

Juan Fernando Tellez
PhD Student
Department of Political Science
Duke University

**Comment by jordan-roberts** [comment]
Howard:  I don't think it should take too long to skim to summaries just to find the sentence about the mediators since not very many crisis have mediation, but I suppose two people doing it might be good just because it would reduce the chance of error.  Any takers?

Juan I: The update won't be out for a while still.  However, Kyle said that either he or David Quinn could share the relevant variables with us as soon as they are complete (which should be much sooner than the full data).  We will likely just need to manually put that data into a dataframe and append it to the end of what we already have.  

Juan II: On the crisisName variable, I wasn't suggesting we remove spaces.  I was trying to point out that we are actually missing spaces in some places!  In the example above, the paste command I originally wrote captures the space between CRISIS and NAME, but not the space between NAME and EXAMPLE.  Maybe it's not a big deal, but would be slightly more readable if all spaces were included somehow.  Maybe checking if a cluster is one character shorter than most indicating an excluded string  "space" at the end?  This probably isn't worth a ton of effort, as we could just manually add the spaces in a few minutes if we really cared.

**Comment by hail2thief** [comment]
Yeah the funny thing is that if you import via SPSS (ICB10) then a simple
paste of the clusters produces a readable version of the crisis:

[image: Inline image 1]

Once I get the STATA version of the data (ICB 12) I could work on making
those names more readable.

BTW, what possible reason was there to originally split up the crisis name
into multiple "clusters"?

On Thu, Sep 24, 2015 at 11:38 AM, Jordan Roberts <notifications@github.com>
wrote:

> Howard: I don't think it should take too long to skim to summaries just to
> find the sentence about the mediators since not very many crisis have
> mediation, but I suppose two people doing it might be good just because it
> would reduce the chance of error. Any takers?
>
> Juan I: The update won't be out for a while still. However, Kyle said that
> either he or David Quinn could share the relevant variables with us as soon
> as they are complete (which should be much sooner than the full data). We
> will likely just need to manually put that data into a dataframe and append
> it to the end of what we already have.
>
> Juan II: On the crisisName variable, I wasn't suggesting we remove spaces.
> I was trying to point out that we are actually missing spaces in some
> places! In the example above, the paste command I originally wrote captures
> the space between CRISIS and NAME, but not the space between NAME and
> EXAMPLE. Maybe it's not a big deal, but would be slightly more readable if
> all spaces were included somehow. Maybe checking if a cluster is one
> character shorter than most indicating an excluded string "space" at the
> end? This probably isn't worth a ton of effort, as we could just manually
> add the spaces in a few minutes if we really cared.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jordan-roberts/International-Crisis-Dynamics/issues/1#issuecomment-142966729>
> .
>



-- 
Best,

Juan Fernando Tellez
PhD Student
Department of Political Science
Duke University

**Comment by jordan-roberts** [comment]
Ok, I just switched to using the SPSS files as well so we're all on the same page.  This is the most recently released version (the 12 in my STATA filename was referring STATA v. 12, since I had to save an older form because STATA v. 13 files can't be read into R yet).  

The SPSS version might be better or worse since the factors are all strings.  For the mediate variable mentioned above, someone needs to look at the cases for which mediate == "MEDIATION OCCURRED" or something like that.

Also, I think the clusters are just a weird legacy thing since the dataset was started in the 70's.  Couldn't save long strings I think.

**Comment by jordan-roberts** [comment]
Also, I realize now that the goal of the script might be entirely clear.  Maybe someone has a better way of working towards the goal than what I started.

End product should be a dataframe with variables similar to: 
1) country (what country, each row should be a "relevant actor")
2) crisis name (what crisis)
3) crisis id
4) crisis start date
5) crisis end date

2-5 should be the same for each country in the same crisis.  The goal is to have this dataframe from which we can extract the countries and dates of events we need.

A country should be a relevant actor to a crisis if 1) they are a crisis actor (have an observation at the actor level), 2) are the primary source of threat for a crisis actor (southv at the actor level), or 3) are the primary crisis mediator, which we'll need to extract from summaries.

**Comment by lguirola** [comment]
guys, I'm currently swamped with stuff until the end of the week, I'll get
back to you on sunday if that is fine for you :-)

2015-09-24 13:44 GMT-04:00 Jordan Roberts <notifications@github.com>:

> Also, I realize now that the goal of the script might be entirely clear.
> Maybe someone has a better way of working towards the goal than what I
> started.
>
> End product should be a dataframe with variables similar to:
> 1) country (what country, each row should be a "relevant actor")
> 2) crisis name (what crisis)
> 3) crisis id
> 4) crisis start date
> 5) crisis end date
>
> 2-5 should be the same for each country in the same crisis. The goal is to
> have this dataframe from which we can extract the countries and dates of
> events we need.
>
> A country should be a relevant actor to a crisis if 1) they are a crisis
> actor (have an observation at the actor level), 2) are the primary source
> of threat for a crisis actor (southv at the actor level), or 3) are the
> primary crisis mediator, which we'll need to extract from summaries.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jordan-roberts/International-Crisis-Dynamics/issues/1#issuecomment-143000841>
> .
>



-- 

LMG

**Comment by janvogler** [comment]
Hey guys,

I would suggest to allocate the tasks on Monday.

Since I'm not familiar with the dataset it would be better to have the opportunity to ask questions if things are unclear when we distribute the tasks. Also, this way we could give some proper instruction on whatever is necessary to do.

I hope that would be okay. Please let me know your thoughts.

Best,
Jan

**Comment by jordan-roberts** [comment]
Sure, Jan.  But if we do that, we'll need a much more concrete set of tasks.  The ones I put above were mostly pretty minor (except identifying the primary mediators).  I'll try to think about stuff before Monday.

**Comment by jasondtodd** [comment]
I'm adding the mediators, be they primary or otherwise, state actors or otherwise, to the Scope Conditions Script *as comments*. How these things are coded can be a next step (still feels above my pay grade, not being nearly as familiar with the data as some of you).

**Comment by jasondtodd** [comment]
I added some code to Scope Conditions Script to assign sideA and sideB to the actors, and it works for all but the last crisis (Chad-Sudan), because the source of threat for both is some third, unnamed actor.

However, I'm not sure how it will work after all of the other rows are added to the data frame, with mediators and whatnot.
